gensim/models/utils_any2vec.py

```diff
@@ -176,14 +176,64 @@ def _load_word2vec_format(cls, fname, fvocab=None, binary=False, encoding='utf8'
     datatype : type, optional
         (Experimental) Can coerce dimensions to a non-default float type (such as `np.float16`) to save memory.
         Such types may result in much slower bulk operations or incompatibility with optimized routines.)
+    binary_chunk_size : int, optional
+        Size of chunk in which binary files are read. Used mostly for testing. Defalut value 100 kB.
 
     Returns
     -------
     object
         Returns the loaded model as an instance of :class:`cls`.
 
     """
+
+    def __add_word_to_result(result, counts, word, weights):
+        word_id = len(result.vocab)
+        if word in result.vocab:
+            logger.warning("duplicate word '%s' in %s, ignoring all but first", word, fname)
+            return
+        if counts is None:
```

Comment:
You're making the reader do a bit more reading then necessary. You could save your future readers some eyeball work by doing:

```python
if counts is None:
    # most common scenario: no vocab file given. just make up some bogus counts, in descending order
    word_count = vocab_size - word_id
elif word in counts:
    # use count from the vocab file
    word_count = counts[word]
else:
    logger.warning("vocabulary file is incomplete: '%s' is missing", word)
    word_count = None

result.vocab[word] = Vocab(index=word_id, count=word_count)
```

If it does not affect performance, you could also replace the elif-else with a try-catch, as that'd be more Pythonic.

Reply:
This was unchanged by my patches, however, your version of the code definitely reads better.
Included.